content/docs/userspace/http-api-guide.md

```diff
@@ -0,0 +1,649 @@
+do it immediately after instantiating it:
+
+```javascript
+const api = new Urbit("");
+api.ship = window.ship;
+```
+
+### Channels
+
+With the exception of scries and threads, all communication with Eyre happens
+through its channel system.
+
+The `Urbit` object will generate a random channel ID that looks like
+`1646295453-e1bdfd`, and use a path of `/~/channel/1646295453-e1bdfd` for
+channel communications. Pokes will and subscription requests will be sent to
+that channel. Responses and subscription updates will be sent out to you on that
+channel too.
+
+Eyre sends out updates and responses on an [SSE] (Server Sent Event) stream for
+the channel. The `Urbit` object handles this internally with an `eventSource`
+object, but you won't deal with it directly. Eyre requires all events it sends
+out be acknowledged by the client, and will eventually close the channel if
+enough unacknowledged events accumulate. The `Urbit` object handles event
+acknowledgement automatically, so you don't need to worry about this.
+
+Since the `Urbit` object handles channels internally, you don't typically need
+to deal with them at all. There is, however, the [`delete()`](#delete-channel)
+function if you want to close the whole channel and end all subscriptions at
+once.
+
+Eyre automatically creates a channel when a poke or subscription request is
+first sent to `/~/channel/[a new channel ID]`. If your web app is served
+external to the ship and you use the `authenticate` function [described
+below](#authentication), the function will automatically send a poke and open
+the channel. If your web app is served directly from the ship and you use
+`Urbit` class constructor, it won't open the channel right away. Instead, the
+channel will be opened whenever you first send a poke or subscription request.
+
+### Authentication
+
+**If your front-end is served directly from the Urbit ship, this step can be
+skipped.** If your web app is served externally to the ship, you must
+authenticate and obtain a session cookie before commencing communications with
+the ship.
+
+The `Urbit` class in `http-api` includes an `authenticate` function which does
+the following:
+
+1. Login to the user's ship with their `code` and obtain a session cookie.
+2. Generate a random channel ID for the connection.
+3. Poke the user's ship and print "opening airlock" in the dojo to initialize
+   the channel.
+
+The `authenticate` function takes four arguments in an object: `ship`, `url`,
+`code` and `verbose`:
+
+| Argument  | Type      | Description                                                                            | Example                           |
+| --------- | --------- | -------------------------------------------------------------------------------------- | --------------------------------- |
+| `ship`    | `string`  | The ship ID (`@p`) without the leading `~`.                                            | `"sampel-palnet"` or `"zod"`      |
```

Comment:
Not really for *this* guide but would be really nice to make the JS library agnostic to the leading ~.